src/lib/uploadFigmaDesigns.ts

```diff
@@ -0,0 +1,41 @@
+import { Context } from "../types.js";
+import chalk from "chalk";
+import { mapIdToName } from "./utils.js";
+import constants from "./constants.js";
+
+export default async (ctx: Context): Promise<string> => { // Return type changed to string[]
+  const figmaConfigs = ctx.figmaDesignConfig.figma_config;
+  let results = "";
+
+  for (const config of figmaConfigs) {
+    const figmaToken = process.env.FIGMA_TOKEN;
+
+    let queryParams = "";
+    if (config.figma_ids && config.figma_ids.length > 0) {
+      const fileIds = config.figma_ids.join(",");
+      queryParams += `&ids=${fileIds}`;
+    }
+
+    let authToken = "";
+    if (process.env.LT_USERNAME != null && process.env.LT_ACCESS_KEY != null) {
+      authToken = `Basic ${Buffer.from(`${process.env.LT_USERNAME}:${process.env.LT_ACCESS_KEY}`).toString("base64")}`
+    }
+
+    // Construct the request body
+    const requestBody = {
```

Comment:
pass necessary variables to client function and create the request body there. Also, project token is added to every reqeust so need for it again.